admiral/pkg/clusters/util.go

```diff
@@ -14,7 +14,23 @@ import (
 	k8sV1 "k8s.io/api/core/v1"
 )
 
-func GetMeshPorts(clusterName string, destService *k8sV1.Service,
+func GetMeshPortAndLabelsFromDeploymentOrRollout(
+	cluster string, serviceInstance *k8sV1.Service,
+	deploymentsByCluster map[string]*k8sAppsV1.Deployment,
+	rolloutsByCluster map[string]*argo.Rollout,
+) (portsByProtocol map[string]uint32, labels map[string]string) {
+	if len(deploymentsByCluster) > 0 && deploymentsByCluster[cluster] != nil {
+		deployment := deploymentsByCluster[cluster]
+		return GetMeshPortsForDeployment(cluster, serviceInstance, deployment), deployment.Labels
+	}
+	if len(rolloutsByCluster) > 0 && rolloutsByCluster[cluster] != nil {
+		rollout := rolloutsByCluster[cluster]
+		return GetMeshPortsForRollout(cluster, serviceInstance, rollout), rollout.Labels
+	}
+	return nil, nil
+}
+
+func GetMeshPortsForDeployment(clusterName string, destService *k8sV1.Service,
 	destDeployment *k8sAppsV1.Deployment) map[string]uint32 {
 	var meshPorts = destDeployment.Spec.Template.Annotations[common.SidecarEnabledPorts]
```

Comment:
@nirvanagit , should we add a `nil` check here on `destDeployment`?

Reply:
Yes absolutely.